### PR TITLE
Add github workflow running jenkins job

### PR DESCRIPTION
**Pull request by sylwiaszunejko**
This workflow pushes scylla-bench docker image using gocql version from the PR and triggers jenkins job that uses this image of scylla-bench.

### CODE REVIEW

**Comment by sylwiaszunejko** [comment]
New workflow won't be running in CI in this PR, because it uses `pull_request_target` event. In this PR we can see how it behaves: https://github.com/scylladb/gocql/pull/255. 

Things to be changed:
- change jenkins_client fork to scylladb
- change user from me to some official jenkins user
- move jenkins job to official location

**Review by dkropachev — CHANGES_REQUESTED** [review]

File: .github/workflows/extended-ci-asymmetric-cluster.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        uses: actions/checkout@v2
+        with:
+          repository: scylladb/scylla-bench
+          path: scylla-bench
+
+      - name: Checkout GoCQL PR Repository
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          ref: ${{ github.event.pull_request.head.ref }}
```
[the remark this answers is not in the document]
Comment (sylwiaszunejko) [reply .github/workflows/extended-ci-asymmetric-cluster.yml:38]: it seems to be not, look: https://github.com/scylladb/gocql/pull/255/checks
Or do you mean I should add `path: gocql` in some other place?

File: .github/workflows/extended-ci-asymmetric-cluster.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          go mod edit -replace github.com/gocql/gocql=../gocql
+          go mod tidy
+
+      - name: Build and push Scylla-bench Docker Image
+        run: |
+          cd scylla-bench
+          export SCYLLA_BENCH_VERSION=tags/v0.1.22
+          export NAME="${{ github.event.pull_request.head.ref }}"
+          export SCYLLA_BENCH_DOCKER_IMAGE=scylladb/gocql-extended-ci:scylla-bench-${NAME}
+          docker build -f ../gocql/.github/workflows/Dockerfile . -t ${SCYLLA_BENCH_DOCKER_IMAGE} --build-arg version=$SCYLLA_BENCH_VERSION
```
[the remark this answers is not in the document]
Comment (sylwiaszunejko) [reply .github/workflows/extended-ci-asymmetric-cluster.yml:53]: https://github.com/scylladb/scylla-bench/pull/143

**Comment by sylwiaszunejko** [comment]
I went with @Lorak-mmk advice https://github.com/scylladb/gocql/issues/204#issuecomment-2352727699 
Now the changes are here:  https://github.com/scylladb/gocql/pull/255 and here we can see the new workflow running: https://github.com/scylladb/gocql/pull/263
Please move to these two PR with all discussions

**Pull request closed without merging**